tensorflow/core/common_runtime/sycl/sycl_device.cc

```diff
@@ -100,6 +103,64 @@ Status SYCLDevice::Sync() {
   }
 }
 
+void GSYCLInterface::select_specific_platform_vendor_id(GSYCLInterface::device_list_t& device_list, bool& found_device) {
+  auto platform_vendor_cstr = std::getenv("TENSORFLOW_SYCL_USE_PLATFORM_NAME_VENDOR_ID");
+  if (!platform_vendor_cstr)
+    return;
+
+  std::string platform_vendor_str(platform_vendor_cstr);
+  auto sep = platform_vendor_str.rfind(':');
+  if (sep == std::string::npos || sep == 0 || sep == platform_vendor_str.size() - 1) {
+    LOG(WARNING) << "TENSORFLOW_SYCL_USE_PLATFORM_NAME_VENDOR_ID ignored, "
+                    "expected format <platform_name>:<vendor_id>";
+    return;
+  }
+
+  std::string platform_name(platform_vendor_str, 0, sep);
+  auto vendor_id = strtol(std::string(platform_vendor_str, sep + 1).c_str(), NULL, 0);
+
+  // Select the device and remove it from device_list to not select it anymore
+  select_first_device(device_list, found_device, [&](const cl::sycl::device& d) {
+    if (d.get_info<cl::sycl::info::device::vendor_id>() != vendor_id)
+      return false;
+
+    auto device_platform_name = d.get_platform().get_info<cl::sycl::info::platform::name>();
+    if (device_platform_name.back() == '\0') // Do not compare the last null character
+      device_platform_name.erase(device_platform_name.size() - 1);
+    return device_platform_name == platform_name;
+  });
+  if (!found_device) {
+    LOG(WARNING) << "TENSORFLOW_SYCL_USE_PLATFORM_NAME_VENDOR_ID ignored, "
+                    "could not find a supported device with platform name \""
+                 << platform_name << "\" and vendor id " << vendor_id;
+  }
+}
+
+void GSYCLInterface::select_specific_device_type(GSYCLInterface::device_list_t& device_list, bool& found_device) {
+  auto device_type = std::getenv("TENSORFLOW_SYCL_USE_DEVICE_TYPE");
```

Comment:
There is slightly bigger question that we need to ask here and perhaps @benoitsteiner could help out here.. 
How should we handle situation when users system has multiple OpenCL platforms? 
What is desired granularity - should user be able to place Ops on different devices? Or should we pass desired device to the session? Or shall we stick to rigid priority list? OpenCL: Accelerators, GPU, CPU, Host?